api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -156,6 +158,19 @@ public static boolean selectsPartitions(
         caseSensitive);
   }
 
+  public static boolean isPartitionCol(BoundGroupBy expr, Table table) {
+    return table.specs().values().stream().allMatch(spec -> isPartitionCol(expr, spec));
```

Comment:
This check seems overly broad. The aggregate can be pushed down if the partition column exists in all specs that have matching data files. The table may start out unpartitioned and have partitions added later, and then have the initial data rewritten. We want to test whether the specs that are actually used meet the requirement and stop aggregation if there's a problem.